spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/core/Mqttv3ClientManager.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2022-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.mqtt.core;
+
+import java.time.Instant;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.eclipse.paho.client.mqttv3.IMqttAsyncClient;
+import org.eclipse.paho.client.mqttv3.IMqttDeliveryToken;
+import org.eclipse.paho.client.mqttv3.MqttAsyncClient;
+import org.eclipse.paho.client.mqttv3.MqttCallback;
+import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.paho.client.mqttv3.MqttMessage;
+
+import org.springframework.integration.context.IntegrationObjectSupport;
+import org.springframework.integration.support.management.ManageableLifecycle;
+
+public class Mqttv3ClientManager extends IntegrationObjectSupport implements ClientManager<IMqttAsyncClient>,
+		ManageableLifecycle, MqttCallback {
+
+	private AtomicReference<ScheduledFuture<?>> scheduledReconnect;
+
+	private final MqttConnectOptions connectOptions;
+
+	private final String clientId;
+
+	private IMqttAsyncClient client;
+
+	public Mqttv3ClientManager(MqttConnectOptions connectOptions, String clientId) throws MqttException {
```

Comment:
Let's see if we can get rid of this exception, when we don't create a client instance over here!
We still going to have the respective logic in the `start()`. 
It is definitely better do not do this client instantiation in the ctor anyway: I see they have this over there `this.persistence.open(clientId, serverURI);`, which is, essentially, an interaction with low-level resource, therefore better to postpone to the `start()`.